### DISCUSSION TITLE
[Feature Request] Disable animations globally

### DISCUSSION THREAD

**Discussion by xavierLowmiller** [airbnb/lottie-ios#1918] [Ideas]
# Description

In our app, we do a lot of [snapshot testing](https://github.com/pointfreeco/swift-snapshot-testing).
To keep these deterministic, we disable the Lottie animations by hand, which works :+1:.

It would be much easier, though, if Lottie looked at `UIView.areAnimationsEnabled` so that animations could be enabled or disabled globally.

If you're interested, I can try and submit a pull request.

## Check these before submitting:
- [x] The issue doesn't involve an [Unsupported Feature](https://github.com/airbnb/lottie-ios/blob/master/README.md#supported-after-effects-features)
- [x] This issue isn't related to another open issue

## This issue is a:
- [ ] Non-Crashing Bug (Visual or otherwise)
- [ ] Crashing Bug
- [x] Feature Request
- [ ] Regression (Something that once worked, but doesn't work anymore)

## Which Version of Lottie are you using?

Lottie 3.0

## What Platform are you on? 
- [ ] MacOS
- [x] iOS

## What Language are you in?
- [x] Swift
- [ ] Objective-C

**Comment by kwstasna** [comment]
@xavierLowmiller how do you disable the animations while you take the snapshots?

**Comment by xavierLowmiller** [comment]
Just set `UIView.setAnimationsEnabled(false)`

**Comment by kwstasna** [comment]
@xavierLowmiller ah ok! Do you know how to display only the final frame so that the test don’t have to wait for the animation to finish ?

**Comment by xavierLowmiller** [comment]
Oh sorry, you mean how we disable the Lottie animations?

I can't take a look how we exactly did it because I've switched jobs and don't have access to the code base.

As far as I remember, we just called `stop()` on the view, and maybe set the `animationProgress` to some deterministic value.

**Comment by amja** [comment]
Was there any reason this wasn't taken up? Looks like Lottie Android has incorporated similar behaviour. Would be super useful for many people I am sure!

**Comment by calda** [comment]
We added support for reduced motion in https://github.com/airbnb/lottie-ios/pull/2110

**Reply by calda** [reply]
Ah, we don't handle `UIView.areAnimationsEnabled` though.

**Reply by amja** [reply]
I'm glad reduced motion is supported as that's a much more important use case but yes, would be wonderful if we could do something similar for `UIView.areAnimationsEnabled`.

**Reply by calda** [reply]
Anyone interested in working on a PR? Happy to review one.

**Reply by amja** [reply]
I could definitely do it but I had a moment of doubt and would like to get some consensus. Though it's not clearly defined in the docs, it seems like `UIView.setAnimationsEnabled(:)`'s purpose is mainly to disable UIKit lifecycle animation like view controller presentation etc. It doesn't, for example, stop a `UIActivityIndicatorView` from spinning if you call the `startAnimating()` method on it.

Though it definitely would be convenient to have an ability to globally disable Lottie animation, I worry that hooking into this system might not be totally appropriate.

**Reply by calda** [reply]
We could use an approach similar to https://github.com/airbnb/lottie-ios/pull/2110, where it uses `UIAccessibility.isReduceMotionEnabled` [by default](https://github.com/airbnb/lottie-ios/pull/2110/files#diff-d48b4be67b645a31a872939f3e566646844bc2f8d17e73cd1c53aa298defdfb4R105) but in a way that lets the consumer override this with [some other configuration](https://github.com/airbnb/lottie-ios/pull/2110/files#diff-d48b4be67b645a31a872939f3e566646844bc2f8d17e73cd1c53aa298defdfb4R20) if they want.
